src/aleph/http/schema.clj

```diff
@@ -0,0 +1,46 @@
+(ns ^:no-doc aleph.http.schema
+  (:require [malli.core :as m]))
+
+(def server-port [:maybe :int])
+(def server-name [:maybe :string])
+(def remote-addr [:maybe :string])
+(def uri [:maybe :string])
+(def query-string [:maybe :string])
+(def scheme [:enum :http :https])
+(def request-method :keyword)
+(def content-type [:maybe [:or :string :keyword]])
+(def content-length [:maybe :int])
+(def character-encoding [:maybe :string])
+
+(def ring-request [:map
+                   [:request-method request-method]
+                   [:server-port {:optional true} server-port]
+                   [:server-name {:optional true} server-name]
+                   [:remote-addr {:optional true} remote-addr]
+                   [:uri {:optional true} uri]
+                   [:query-string {:optional true} query-string]
+                   [:scheme {:optional true} scheme]
+                   [:content-type {:optional true} content-type]
+                   [:content-length {:optional true} content-length]
+                   [:character-encoding {:optional true} character-encoding]])
```

Comment:
What about `:body`?

Reply:
What would you suggest? It's optional, for starters, but the real issue is it has a ton of representations once middleware gets involved: strings, byte arrays, Clojure maps read from JSON, Manifold streams, InputStreams, etc.

Should we enumerate them? It might be useful when testing middleware to check the contract of what they return, but that will take a bit. For now, we could use `:any` to show we haven't forgotten about it.

Reply:
Added `:body`.